WordPress/gutenberg
# Add a description key to theme.json style variations

## What?

- Adds an optional description key to theme.json, theme-i18n.json, and theme.json schema.
- The description is added to the aria label of the style variation container. If the style variation has no description, only the title is used.

Closes https://github.com/WordPress/gutenberg/issues/44960

## Why?

This is intended to be an accessibility improvement. Style variations are mostly visual, and there needs to be a text alternative that describes them.
One purpose is to use the text description to improve the style variation preview in the WordPress.org theme directory.

## Testing Instructions
Install the theme Twenty Twenty-Three.
In your code editor, locate and open the Aubergine.json style variation in the themes `styles `folder.
Add the new description key and value as a top-level key in the JSON file:
`"description": "Your description here"`
```
{
	"$schema": "https://schemas.wp.org/trunk/theme.json",
	"version": 2,
	"title": "Aubergine",
	"description": "This is the test description"
```
Activate the theme.
Go to Appearance, Editor.
Activate your screen reader if you haven't already.
In the Navigation Sidebar, select Styles from the menu.
Using the tab key, navigate past the Back button, the Stylebook button, and the Edit Styles button.
Navigate past the first style variation button, which is called Default.
The second button should announce the name "Aubergine" followed by the description you entered.

Next, please test the style variations in the edit mode:
Use the Shift + tab to navigate backward until you reach the Edit Styles button. Activate the button.
Tab forward until you reach the Stylebook button; this button means you have reached the Styles sidebar.
Tab forward past the Style Actions button until you reach Browse Styles. Activate this button.
Navigate past the first style variation button, which is called Default.
The second button should announce the name "Aubergine" followed by the description you entered.

## Screenshots or screencast 
https://github.com/WordPress/gutenberg/assets/7422055/b1355998-c1e5-4036-a3b0-7a2c8df72797
